### PR TITLE
[WiP] Replace unwrap() with percolated error handling

### LINKED ISSUE #24: Remove all `unwraps` and use proper error propagation instead
If you search for `unwrap` in the code basis, you will find far too many.
Unwraps make sense only in main functions.

In most places, they are present inside closures used in helper functions of iterators. In those cases, they should be replaced by first a map that changes `Iterator<T>` to `Iterator<Result<T, Error>>` and then a collect to transform it to `Result<Vec<T>, Error>`.

In the demo binaries, instead of unwrap, we should log the error and continue iterating.

### PR DESCRIPTION
**Pull request by SamuelMarks**
Closes #24

**WiP**

### CODE REVIEW

**Comment by SamuelMarks** [comment]
I've ignored a bunch of errors, because I was thinking that we don't want nodes to crash.
Not sure if you want to use the `log` crate or similar to actually log the error before ignoring it in the next iterator functor.

I didn't remove the `unwrap`s within the test modules or `main` functions, because I think it's fine for panics to happen here.

Not sure how to remove the `unwrap` from threads.

Currently debugging the last few remaining compilation errors, then ready for @AgustinCB to review.

**Comment by AgustinCB** [comment]
We already use `log`, so feel free to log them or open a new issue to handle that

File: lachesis-rs/src/round.rs [hunk 1]
```diff
@@ -37,6 +38,7 @@ mod tests {
     #[test]
     fn it_should_correctly_get_all_witnesses() {
         let mut round = Round::new(0);
+        let a = round.clone();
```
Comment (rishflab) [inline lachesis-rs/src/round.rs]: remove

**Comment by AgustinCB** [comment]
Let me know guys when is this ready for review

**Comment by SamuelMarks** [comment]
@AgustinCB can you merge this? - The rest of the codebase will move to far in the wrong direction without this.

**Pull request merged**